mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala

```diff
@@ -562,6 +573,23 @@ private[ml] object SummaryBuilderImpl extends Logging {
 
       Vectors.dense(currL1)
     }
+
+    /**
+     * Sum of each dimension
+     */
+    def sum: Vector = {
+      require(requestedMetrics.contains(Sum))
+      require(totalWeightSum > 0, s"Nothing has been added to this summarizer.")
+
+      val realSum = Array.ofDim[Double](n)
+      var i = 0
+      val len = currMean.length
+      while (i < len) {
+        realSum(i) = currMean(i) * weightSum(i)
+        i += 1
```

Comment:
Please avoid using mutable values, use foldLeft for example to solve this.

Reply:
I will change that.